src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -161,13 +170,85 @@ static unsigned uct_cuda_copy_iface_progress(uct_iface_h tl_iface)
     unsigned max_events = iface->config.max_poll;
     unsigned count;
 
-    count = uct_cuda_copy_progress_event_queue(&iface->outstanding_d2h_cuda_event_q,
+    count = uct_cuda_copy_progress_event_queue(iface, UCT_CUDA_COPY_D2H_STREAM,
                                                max_events);
-    count += uct_cuda_copy_progress_event_queue(&iface->outstanding_h2d_cuda_event_q,
+    count += uct_cuda_copy_progress_event_queue(iface, UCT_CUDA_COPY_H2D_STREAM,
                                                 (max_events - count));
     return count;
 }
 
+static void uct_cuda_copy_common_cb(void *cuda_copy_iface)
+{
+    uct_cuda_copy_iface_t *iface = cuda_copy_iface;
+
+    pthread_mutex_lock(&iface->mutex);
+    iface->cb_counter = (iface->cb_counter == ULLONG_MAX - 1) ?
+                        (iface->cb_counter + 1) : 1;
+    pthread_mutex_unlock(&iface->mutex);
+
+    ucs_assert(iface->async.event_cb != NULL);
+    /* notify user */
+    iface->async.event_cb(iface->async.event_arg, 0);
```

Comment:
@yosefe Can you remind me the block/unblock functions to guard this function?

Reply:
UCS_ASYNC_BLOCK/UCS_ASYNC_UNBLOCK